infrahub_sdk/protocols_generator/generator.py

```diff
@@ -71,29 +56,44 @@ def __init__(self, schema: dict[str, MainSchemaTypesAll]):
         self.sorted_profiles = self._sort_and_filter_models(
             self.profiles, filters=["CoreProfile"] + self.base_protocols
         )
+        self.sorted_templates = self._sort_and_filter_models(
+            self.templates, filters=["CoreObjectTemplate"] + self.base_protocols
+        )
 
     def render(self, sync: bool = True) -> str:
         jinja2_env = jinja2.Environment(loader=jinja2.BaseLoader(), trim_blocks=True, lstrip_blocks=True)
-        jinja2_env.filters["inheritance"] = self._jinja2_filter_inheritance
         jinja2_env.filters["render_attribute"] = self._jinja2_filter_render_attribute
         jinja2_env.filters["render_relationship"] = self._jinja2_filter_render_relationship
+        jinja2_env.filters["syncify"] = self._jinja2_filter_syncify
 
-        template = jinja2_env.from_string(PROTOCOLS_TEMPLATE)
+        template = jinja2_env.from_string(load_template())
```

Comment:
I don't think we need to change this in this PR but I think it would make sense to the templating logic we have elsewhere within the SDK to render these templates. Possibly we should come to a final decision regarding #337 first if we should leave it as async by default or not.